lib/internal/file.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+const {
+  DateNow,
+  NumberIsNaN,
+  ObjectDefineProperties,
+  ObjectPrototypeHasOwnProperty,
+  Symbol,
+  SymbolToStringTag,
+} = primordials;
+
+const {
+  Blob,
+} = require('internal/blob');
+
+const {
+  customInspectSymbol: kInspect,
+  emitExperimentalWarning,
+  kEnumerableProperty,
+  kEmptyObject,
+  toUSVString,
+} = require('internal/util');
+
+const {
+  codes: {
+    ERR_INVALID_THIS,
+    ERR_MISSING_ARGS,
+  },
+} = require('internal/errors');
+
+const {
+  inspect,
+} = require('internal/util/inspect');
+
+const kState = Symbol('kState');
+
+function isFile(object) {
+  return object?.[kState] !== undefined || object instanceof Blob;
+}
+
+class File extends Blob {
+  constructor(fileBits, fileName, options = kEmptyObject) {
+    emitExperimentalWarning('buffer.File');
+
+    if (arguments.length < 2) {
+      throw new ERR_MISSING_ARGS('fileBits', 'fileName');
+    }
+
+    super(fileBits, options);
+
+    let lastModified;
+
+    if (ObjectPrototypeHasOwnProperty(options, 'lastModified')) {
+      // Using Number(...) will not throw an error for bigints.
+      lastModified = +options.lastModified;
+
+      if (NumberIsNaN(lastModified)) {
+        lastModified = 0;
+      }
+    } else {
+      lastModified = DateNow();
+    }
+
+    this[kState] = {
+      name: toUSVString(fileName),
+      lastModified,
+    };
```

Comment:
Shouldn't we use private properties, so we don't need those `isFile(this)` checks at all?

```suggestion
    this.#name = toUSVString(fileName);
    this.#lastModified = lastModified;
```

Reply:
I think checking that `this` is a File is still a good idea so we can throw a better error, but I'll change these to private props